cypher-editor-support/src/completion/AutoCompletion.js

```diff
@@ -203,7 +200,10 @@ export class AutoCompletion {
       .map(t => t.complete(types, query))
       .reduce((acc, items) => [...acc, ...items], [])
       .filter(completionItemFilter);
-
+    if (text.startsWith('$')) {
```

Comment:
Move this logic inside 'filterText' function.

The idea is that this 'filterText' function will be capable of checking text and doing necessary transformation.
So, kinda encapsulates the internal logic.